### DISCUSSION TITLE
Questions about 4.x-compat as a package maintainer

### DISCUSSION THREAD

**Discussion by TiZ-HugLife** [Palakis/obs-websocket#909] [General]
Hi there. I'm the maintainer for the Flathub package for obs-websocket. I'm sure that after 5.0.0 is out and bundled with OBS, the package won't be particularly needed anymore, but I wanted to get ready for the future. Right now, the Flatpak has 4.9.1 in it. I got the email for the release of 4.9.1-compat, but the release email implied that there would be configuration breakages for users, and so I was reticent to migrate to it because I didn't want to break existing setups. I wasn't sure how I would be able to get the messaging out to users of the package.

However, since the release of that tag, the 4.x-compat branch has had changes to revert the default port to 4444 and start reading 4.x configs again. A corresponding tag has not been made yet. Will 5.x use a different default port from 4.x, and if so, would you say it's safe to use the head of the 4.x-compat branch for the Flatpak? If so, then I shouldn't need to tell users anything, and they can simply continue on without any sort of intervention after the update.

**Comment by tt2468** [comment]
The compat branch is basically 4.9.1 but with the binary renamed to `obs-websocket-compat.so`. We did this so that you can have 5.x and 4.x protocol support on the same OBS instance when 5 comes out.

We had originally planned to make both protocols use the same port, and expect the applications to differentiate between the two, but we scrapped that plan. The new plan is: 4.x protocol uses 4444, 5.x protocol uses 4455

The reason for the default port changes in compat was to allow both WebSocket server instances to start up without port conflicts. This was reverted and now defaults to 4444 again.

We pulled the compat release from github as a result of people getting confused a lot. We will be including the compat binaries in the download section of the eventual 5.0.0 release.

Packaging-wise, how you want to implement that is up to you. I'm not familiar with Flatpak to be able to provide many suggestions, sorry.

**Reply by TiZ-HugLife** [reply]
Thanks very kindly for your reply! My goal is basically future-proofing and making sure that so doing impacts current users as little as possible. Since the default port and configuration changes were reverted, does that mean there are essentially no code changes between 4.9.1 and 4.x-compat aside from the name of the resulting binary?

If that is the case, and since you are planning to include the compatibility binaries with 5.0 stable anyways, then maybe a good plan is to just do nothing for now, and then when 5.0 comes out, build both 5.0 and 4.x-compat until OBS starts bundling obs-websocket, at which point the Flatpak for the plugin shifts to being only for 4.x compatibility.

@GeorgesStavracas, you were heavily involved in getting Flatpak supported in OBS Studio's upstream. Does this plan sound okay to you?

**Reply by GeorgesStavracas** [reply]
Sounds like a good plan to me :+1:
